apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -848,6 +848,20 @@ async function legacyTraverseQuestionAndExecuteFunctions(phase, codeCaller, data
   };
 }
 
+/**
+ * @param {unknown} val
+ * @return {{ score: number, weight: number }}
+ */
+function getPartialScoreValues(val) {
+  const obj = /** @type {Record<string, unknown>} */ (
+    typeof val === 'object' && val != null ? val : {}
+  );
+  return {
+    score: typeof obj.score === 'number' ? obj.score : 0,
+    weight: typeof obj.weight === 'number' ? obj.weight : 1,
```

Comment:
This is a slight change in behavior. Previously, a non-number value would be silently used in the math, so e.g. if `score` was `"foo"`, `total_weight` could be `"0foo"`. Now, we silently reject non-numbers and use the as the default value instead.

Maybe we should error more loudly about that? I'm inclined to save that for some future instructor-only warning system.

Reply:
I think saving it for the future, and marking that as an issue is good. I suspect this may break some questions.